rust/module.rs

```diff
@@ -244,18 +270,59 @@ pub fn module(ts: TokenStream) -> TokenStream {
             &param_name,
             &param_description,
         ));
+        let param_type_internal = match param_type.as_ref() {
+            "str" => "*mut u8",
+            "invbool" => "bool",
+            _ => &param_type,
+        };
+        let param_default = match param_type.as_ref() {
+            "str" => format!("b\"{}\0\" as *const u8 as *mut u8", param_default),
+            _ => param_default,
+        };
+        let read_func = match param_type.as_ref() {
+            "str" => format!(
+                "
+                    fn read(&self) -> Result<&str, core::str::Utf8Error> {{
+                        unsafe {{
+                            let mut count = 0;
+                            while *__{name}_{param_name}_value.add(count) != b'\0' {{
+                                count += 1;
+                            }}
```

Comment:
Is there anything in `core` or `alloc` that finds out the length of a C-string? Or perhaps even to build directly the raw parts? If not, we should probably add it to our utilities somewhere in `rust/kernel` (and that implementation should probably call the kernel's `strlen()` since it is `unsafe` anyway).

Reply:
It looks like the various `CStr`/`CString` stuff is just in `std`. I added a function to `kernel::c_types` using `strlen`.